components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/CacheBackedIdPMgtDAO.java

```diff
@@ -84,6 +85,39 @@ public CacheBackedIdPMgtDAO(IdPManagementDAO idPMgtDAO) {
 		return idPMgtDAO.getIdPsSearch(dbConnection, tenantId, tenantDomain,
 				filter);
 	}
+
+    /**
+     * Get all basic identity provider information for a matching filter.
+     *
+     * @param tenantId             tenant Id of the identity provider.
+     * @param expressionConditions filter value list for IdP search.
+     * @param limit                limit per page.
+     * @param offset               offset value.
+     * @param sortOrder            order of IdP ASC/DESC.
+     * @param sortBy               the attribute need to sort.
+     * @return Identity Provider's basic information array.
+     * @throws IdentityProviderManagementException Error when getting list of Identity Providers.
+     */
+    public List<IdentityProvider> getIdPsSearch(int tenantId, List<ExpressionNode> expressionConditions, int limit,
+            int offset, String sortOrder, String sortBy) throws IdentityProviderManagementException {
+
+        return idPMgtDAO.getIdPsSearch(tenantId, expressionConditions, limit, offset, sortOrder, sortBy);
+    }
+
+    /**
+     * Get number of IdP count for a matching filter.
+     *
+     * @param tenantId             Tenant Id of the identity provider.
+     * @param expressionConditions filter value list for IdP search.
+     * @return number of IdP count for a given filter.
+     * @throws IdentityProviderManagementException Error when getting count of Identity Providers.
+     */
+    public int getCountOfFilteredIdPs(int tenantId, List<ExpressionNode> expressionConditions)
+            throws IdentityProviderManagementException {
+
+        return idPMgtDAO.getCountOfFilteredIdPs(tenantId, expressionConditions);
```

Comment:
if the 'expressionConditions' is set as null, can we get the unfiltered total IDP count from this function?